lib/src/components/Divider/Divider.tsx

```diff
@@ -0,0 +1,71 @@
+import { assignInlineVars } from "@vanilla-extract/dynamic";
+import {
+  dividerContainerHorizontal,
+  dividerContainerVertical,
+  dividerHorizontal,
+  dividerThemeVar,
+  dividerVertical,
+  labelHorizontal,
+  labelVertical,
+} from "./divider.css";
+import { DividerProps } from "./divider.types";
+
+export const Divider = ({
+  type = "solid",
+  label,
+  labelPosition = "center",
+  color = "grey",
+  labelColor = "#000",
+  labelBackground = "#fff",
+  orientation = "horizontal",
+  size = "1px",
+  ...nativeProps
+}: DividerProps) => {
+  const dividerClass =
+    orientation === "horizontal"
+      ? dividerHorizontal({
+          type,
+        })
+      : dividerVertical({
+          type,
+        });
+  const labelClass =
+    orientation === "horizontal"
+      ? labelHorizontal({
+          labelPosition,
+        })
+      : labelVertical({
+          labelPosition,
+        });
```

Comment:
we could use a simple recipie here instead of the logic here?

Reply:
I initially started with a recipe, but it ended up having too many conditions...so to improve the code readability gone with this approach.

Reply:
what i am saying is keep 2 seperate recipies, 1 for the wrapping div and the other one you are already using, will be cleaner this way

Reply:
based on discussion with @saurabhsutar192 recipes will make the code complex so we will keep the current code